src/plot/crosshair.js

```diff
@@ -132,14 +133,19 @@ class Crosshair extends PureRenderComponent {
       marginLeft,
       innerWidth,
       innerHeight} = this.props;
+    let {orientation} = this.props;
     const value = getFirstNonEmptyValue(values);
     if (!value) {
       return null;
     }
     const x = getAttributeFunctor(this.props, 'x');
     const innerLeft = x(value);
 
-    const orientation = (innerLeft > innerWidth / 2) ? 'left' : 'right';
+    // infer orientation from x value if not overridden
+    if (!orientation) {
+      orientation = (innerLeft > innerWidth / 2) ? 'left' : 'right';
```

Comment:
no, since `getDefaultProps` shouldn't access `this.props`. If this http://stackoverflow.com/a/33127059/1214462

I could create an instance method for this called `getOrientation()` that either returns the prop or the inference, which would prevent reassignment in `render()`.